src/index.js

```diff
@@ -1,162 +1,8 @@
-      this.listening = true;
-    }
-  }
-
-  /**
-   * Remove event listeners on unmount if we are still
-   * listening.
-   */
-  componentWillUnmount() {
-      if (this.listening) {
-        this.removeEventListeners()
-      }
-  }
-
-  /**
-   * Removes the `mousemove` and `resize` listeners. May
-   * be called on unmount, or after `onBreach` is called
-   * if the `once` prop is set to `true`
-   */
-  removeEventListeners() {
-    window.removeEventListener('mousemove', this.handleMouseMove);
-    window.removeEventListener('resize', this.handleResize);
-    this.listening = false;
-  }
-
-  /**
-   * An element's position may change on resize, so we recalculate
-   * the ClientRect for the node.
-   * @todo debounce
-   */
-  handleResize = () => {
-    if (this.node) {
-      this.bounds = this.node.getBoundingClientRect();
-    }
-  }
-
-  /**
-   * Called on `mousemove`, using `clientX` and `clientY`
-   * event properties to determine the position of the cursor.
-   * The perimeter is then calculated from the ClientRect and the
-   * `padding` prop. If the mouse is within the perimeter,
-   * and its not already breached (tracked on `this.breached`)
-   * @param {MouseEvent} event mouse event
-   */
-  handleMouseMove = ({ clientX, clientY } : MouseEvent) => {
-    const {initialOffset, props, bounds } = this;
-    /**
-     * There should be no situation where the `mousemove` handler
-     * is called and `bounds` is not calculated, but flow demands
-     * this check since the initial property value is `null` 
-     */
-    if (!bounds) return;
-    const offsetY = window.pageYOffset - initialOffset;
-    const { padding, onBreach, once } = props;
-    const { top, right, bottom, left } = bounds;
-    if (
-        // Cursor is not too far left
-        clientX > (left - padding) &&
-        // Cursor is not too far right
-        clientX < (right + padding) &&
-        // Cursor is not too far up
-        clientY > ((top - offsetY) - padding) &&
-        // Cursor is not too far down
-        clientY < ((bottom - offsetY) + padding)
-    ) {
-      if (this.breached) {
-        return
-      }
-      onBreach()
-      this.breached = true;
-      if (once) {
-        this.removeEventListeners()
-      }
-    } else {
-      this.breached = false;
-    }
-  }
-
-  /**
-   * Ref callback used to populate this.node. If
-   * a render callback is provided then it will be passed
-   * to that callback with the expectation of it then
-   * being passed to an element via the `ref` prop somewhere.
-   * Otherwise the default `span` is rendered and populated.
-   */
-  attachRef = (node: HTMLElement) => {
-    this.node = node
-  }
-
-  render() {
-      const { children } = this.props;
-      return typeof children === 'function'
-        ? children(this.attachRef)
-        : <span ref={this.attachRef}>{children}</span>
-  }
-}
\ No newline at end of file
+export default Perimiter;
```

Comment:
We don't need to export `Perimeter` twice. We can also just re-export the modules directly. Lets change this file to:

```js
export MouseMoveProvider from './MouseMoveProvider';
export default from './Provider';
```